src/detect-bytemath.c

```diff
@@ -0,0 +1,1364 @@
+            bmd->base = DETECT_BYTEMATH_BASE_HEX;
+        } else if (strcmp("oct", tmp_str) == 0) {
+            bmd->base = DETECT_BYTEMATH_BASE_OCT;
+        } else if (strcmp("dec", tmp_str) == 0) {
+            bmd->base = DETECT_BYTEMATH_BASE_DEC;
+        }
+    }
+
+    if (ret > DCE_KW) {
+        res = pcre_copy_substring((char *)arg, ov, MAX_SUBSTRINGS,
+                                  DCE_KW, tmp_str, sizeof(tmp_str));
+        if (res < 0) {
+            SCLogError(SC_ERR_PCRE_GET_SUBSTRING, "pcre_copy_substring failed "
+                       "for byte_math \"dce\" arg");
+            goto error;
+        }
+
+        if (tmp_str[0] != '\0') {
+            bmd->flags |= DETECT_BYTEMATH_FLAG_ENDIAN;
+            bmd->endian = DETECT_BYTEMATH_ENDIAN_DCE;
+        }
+    }
+
+    if (ret > BITMASK_VAL) {
+        res = pcre_copy_substring((char *)arg, ov, MAX_SUBSTRINGS,
+                                  BITMASK_KW, tmp_str, sizeof(tmp_str));
+        if (res < 0) {
+            SCLogError(SC_ERR_PCRE_GET_SUBSTRING, "pcre_copy_substring failed "
+                       "for byte_math \"bitmask\" arg");
+            goto error;
+        }
+
+        if (tmp_str[0] != '\0') {
+            bmd->flags |= DETECT_BYTEMATH_FLAG_BITMASK;
+        }
+
+        /* bitmask value*/
+        res = pcre_copy_substring((char *)arg, ov, MAX_SUBSTRINGS,
+                                  BITMASK_VAL, tmp_str, sizeof(tmp_str));
+        if (res < 0) {
+            SCLogError(SC_ERR_PCRE_GET_SUBSTRING, "pcre_copy_substring failed "
+                   "for bitmask value: \"%s\"", tmp_str);
+            goto error;
+        }
+
+        res = ByteExtractStringUint32(&bmd->bitmask_val, 16, strlen(tmp_str), tmp_str);
+        if (res < 0) {
+            SCLogError(SC_ERR_INVALID_SIGNATURE, "Unable to extract bitmask "
+                       "value: \"%s\"", tmp_str);
+            goto error;
+        }
+
+        /* determine how many trailing 0's are in the bitmask. This will be used
+         * to rshift the value after applying the bitmask
+         */
+        bmd->bitmask_shift_count = 0;
+        if (bmd->bitmask_val) {
+            uint32_t bmask = bmd->bitmask_val;
+            while (!(bmask & 0x1)){
+                bmask = bmask >> 1;
+                bmd->bitmask_shift_count++;
+            }
+        }
+    }
+
+    if (bmd->endian == DETECT_BYTEMATH_ENDIAN_DCE) {
+        switch (bmd->nbytes) {
+            case 1:
+            case 2:
+            case 4:
+                break;
+            default:
+                SCLogError(SC_ERR_INVALID_SIGNATURE, "nbytes must be 1, 2, or 4 "
+                           "when used with \"dce\"; %d is not valid", bmd->nbytes);
+                goto error;
+                break;
+        }
+    }
+
+    switch (bmd->oper) {
+    	case DETECT_BYTEMATH_OPERATOR_LSHIFT:
+    	case DETECT_BYTEMATH_OPERATOR_RSHIFT:
+            /* nbytes has already been validated to be in [1, 10] */
```

Comment:
whats this `[1, 10]`?

Reply:
I'll update the comment. The value has already been validated to be in the range `[1-10]`.